tests_login_with_email_address/test_visibility_of_elements.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+from assertpy import assert_that, soft_assertions, soft_fail
+
+
+def test_visibility(page, locators=None, locators2=None, locators3=None, locators4=None, locators5=None, locators6=None):
+    with soft_assertions():
+
+        page.goto("https://login.pracuj.pl/")
+        if page.locator("[data-test=\"input-email\"]").is_visible():
+            locators = page.locator("[data-test=\"input-email\"]")
+            assert_that(locators).is_true()
+        else:
+            assert_that(locators).described_as('!!!!!!!!!!!!!!!No selector required!!!!!!!!!!!!!!!').is_true()
```

Comment:
Variable `locators` used before being defined. Move `locators = page.locator("[data-test=\"input-email\"]")` before if/else statement to be correct. The same error is in the entire file.